packages/vow/src/watch.js

```diff
@@ -74,58 +67,50 @@ const prepareWatcherKit = (zone, rejectionMeansRetry, watchVow) =>
      */
     (resolver, watcher) => {
       const state = {
-        vow: undefined,
+        vow: /** @type {unknown} */ (undefined),
         resolver,
         watcher,
       };
       return /** @type {Partial<typeof state>} */ (state);
     },
     {
-      vowSetter: {
-        /** @param {any} vow */
-        setVow(vow) {
-          this.state.vow = vow;
-        },
+      /** @type {Required<import('./watch-promise.js').PromiseWatcher>['onFulfilled']} */
+      onFulfilled(value) {
+        const { watcher, resolver } = this.state;
+        if (getVowPayload(value)) {
+          // We've been shortened, so reflect our state accordingly, and go again.
+          this.state.vow = value;
+          watchVow(value, this.self);
+          return undefined;
+        }
+        this.state.watcher = undefined;
+        this.state.resolver = undefined;
+        if (!resolver) {
+          return undefined;
+        } else if (watcher) {
+          settle(resolver, watcher, 'onFulfilled', value);
```

Comment:
I thought it was a mistake to allow `watchPromise` to use variadic arguments.  I am willing to accept a third argument with the same semantics as `watchPromise`'s third.

Reply:
> can you explain why [it] should return a vow that resolves to results from the watcher

For parity with `Promise.prototype` methods like `.then`, and `.catch`'s chaining functionality, to allow composing vows built with watchers with other watchers.

Reply:
> > can you explain why [it] should return a vow that resolves to results from the watcher
> 
> For parity with `Promise.prototype` methods like `.then`, and `.catch`'s chaining functionality, to allow composing vows built with watchers with other watchers.

Do we have a use case for that? We've gotten this far without anything similar in the `watchPromise` provided by liveslots, so I'm inclined to YAGNI it out of vow `watch` as well—especially since there are likely to be subtle differences/complexities (e.g., what happens if an `onFulfilled` method returns the vow returned from the `watch` to which its object was passed?[^1]) and because nothing about names like "onFulfilled" and "onRejected" imply that the return value is relevant.

[^1]: Spoiler: an infinite loop, in contrast to errors from analogous situations with native promises and with `E`.

Reply:
> We've gotten this far without anything similar in the `watchPromise` provided by liveslots, so I'm inclined to YAGNI it out of vow `watch` as well

IAGNI... in the network vat, we have patterns that look like this:

```js
const result = E(obj).method(...args).then(r => processResult(r)).catch(e => handleRejection(e))
// in consumer
result.then(r => dispatchResult(r));
```

Without chaining, we'd need to create an amalgam of all of those handlers to use as a single watcher.  That seems tricky, especially when those operations are trusted differently.  With chaining, we get:

```js
const processed = watch(E(obj).method(...args), resultProcessingWatcher);
const result = watch(processed, rejectionHandlingWatcher);
// in consumer
watch(result, resultDispatchingWatcher);
```

Handling an infinite vow loop is a different matter... you can create one even without chaining.